app/models/comment.rb

```diff
@@ -303,6 +305,17 @@ def published_article
     errors.add(:commentable_id, "is not valid.") if commentable_type == "Article" && !commentable.published
   end
 
+  def user_mentions_in_markdown
+    # The "comment-mentioned-user" css is added by Html::Parser#user_link_if_exists
+    mentions = Nokogiri::HTML(processed_html).css(".comment-mentioned-user").map do |link|
+      link.text.delete("@").downcase
+    end
```

Comment:
This same code lives in `Mentions::CreateAll#extract_usernames_from_mentions_in_text`. I explicitly not to extract it because it seemed a bit preemptive (read: rule of three), but if we find ourselves having to do this type of check again, I think it would make sense to extract it.

Reply:
I don't think we need to process the mentions, do we?

```ruby
mentions_count = Nokogiri::HTML(processed_html).css(".comment-mentioned-user").size
return if mentions_count <= MAX_USER_MENTIONS
```

Note that I switched the condition because I found it easier to read, feel free to leave it as-is.

Reply:
Oh, of course 🤦🏽‍♀️ We totally don't! Good catch, I didn't feel great about mapping through the markdown here as a part of this validation but thought that we had to for some reason 😅  So glad that we don't :)